### ISSUE TITLE
eth_signTypedData example

### ISSUE THREAD

**Issue by Alko89** [xf00f/web3x#35]
Please provide an example for eth_signTypedData method. I'm trying to use your library instead of web3 and so far it works great!
However I'm trying to implement signTypedData and don't know exactly how. I'm getting Method not found error. Having an example would help a lot.

**Comment by xf00f** [comment]
@Alko89 Could this be due to the provider you are using not supporting `eth_signTypedData`? Usage should be as simple as calling it with `TypedSigningData` as per the function signature. If you can provide an example of what you're trying that would help diagnose.

**Comment by Alko89** [comment]
Yes I think this was also the case. I ended up calling the providers send function directly.

However when looking at the code I noticed dataToSign type is defined as `dataToSign: { type: string; name: string; value: string }[]` which would make it incompatible with EIPs structure (I might be wrong since I'm new to ts).
You can have a look at the sample structure in EIP712 example: https://github.com/ethereum/EIPs/blob/master/assets/eip-712/Example.js

**Comment by xf00f** [comment]
@Alko89 To be honest I implemented this very quickly to get it working in MetaMask a few months back. I wouldn't be surprised if it's now broken.

I will leave the issue open but will probably not get round to doing a proper job on this for a few weeks. I'll gladly accept a PR if you have the time yourself.

**Comment by hickscorp** [comment]
Is there any movement on this issue? We're trying to understand how to sign data using `Web3x`.

**Comment by Alko89** [comment]
Unfortunately I'm not working on the project I have been using web3x for and am currently not involved in Ethereum development.

**Comment by hickscorp** [comment]
We were able to get this to work:

```
// Here, eth is an instance of Web3x/Eth...
eth.signTypedData(state.account, data)
```
